runtime/src/main/java/io/micronaut/scheduling/executor/UserExecutorConfiguration.java

```diff
@@ -37,18 +41,53 @@
      */
     public static final int AVAILABLE_PROCESSORS = Runtime.getRuntime().availableProcessors();
 
+    protected String name;
     // needs to be protected to allow for nThreads to be set in config
     @SuppressWarnings("WeakerAccess")
-    protected Integer nThreads = AVAILABLE_PROCESSORS * 2;
-    private ExecutorType type = ExecutorType.SCHEDULED;
-    private Integer parallelism = AVAILABLE_PROCESSORS;
-    private Integer corePoolSize = AVAILABLE_PROCESSORS * 2;
+    protected Integer nThreads;
+    private ExecutorType type;
+    private Integer parallelism;
+    private Integer corePoolSize;
     private Class<? extends ThreadFactory> threadFactoryClass;
 
+    /**
+     * Private Constructor.
+     *
+     * @param name The name
+     */
+    private UserExecutorConfiguration(@Parameter String name) {
+        this(name, null, null, null, null, null);
+    }
+
     /**
      * Default Constructor.
+     *
+     * @param name the name
+     * @param nThreads number of threads
+     * @param type the type
+     * @param parallelism the parallelism
+     * @param corePoolSize the core pool size
+     * @param threadFactoryClass the thread factory class
      */
-    protected UserExecutorConfiguration() {
+    @Inject
+    protected UserExecutorConfiguration(@Nullable @Parameter String name,
```

Comment:
Are you certain this works? I believe this has to be annotated with `@ConfigurationInject`?